wcfsetup/install/files/lib/system/form/builder/field/UploadFormField.class.php

```diff
@@ -0,0 +1,251 @@
+	 * @return      UploadFile[]
+	 */
+	public function getValue() {
+		$this->registerField();
+		
+		return UploadHandler::getInstance()->getFilesByFieldId($this->getId());
+	}
+	
+	/**
+	 * Returns the removed files for the field. 
+	 * 
+	 * @param       bool    $processFiles
+	 * @return      UploadFile[]
+	 */
+	public function getRemovedFiles($processFiles = false) {
+		$this->registerField();
+		
+		return UploadHandler::getInstance()->getRemovedFiledByFieldId($this->getId(), $processFiles);
+	}
+	
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function getObjectProperty() {
+		return null;
+	}
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function readValue() {
+		$this->registerField(); 
+		
+		return $this;
+	}
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function validate() {
+		parent::validate();
+		
+		if (empty($this->getValue())) {
+			if ($this->isRequired()) {
+				$this->addValidationError(new FormFieldValidationError('empty'));
+			}
+		}
+		
+		if ($this->getMinimum() !== null && count($this->getValue()) < $this->getMinimum()) {
+			$this->addValidationError(new FormFieldValidationError(
+				'minimum',
+				'wcf.form.field.upload.error.minimum',
+				['minimum' => $this->getMinimum()]
+			));
+		}
+		else if ($this->getMaximum() !== null && count($this->getValue()) > $this->getMaximum()) {
+			$this->addValidationError(new FormFieldValidationError(
+				'maximum',
+				'wcf.form.field.upload.error.maximum',
+				['maximum' => $this->getMaximum()]
+			));
+		}
+	}
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function getHtml() {
+		$this->registerField();
+		
+		return parent::getHtml();
+	}
+	
+	/**
+	 * @inheritDoc
+	 * 
+	 * @param       UploadFile[]    $value
+	 * 
+	 * @throws      \InvalidArgumentException       if the value is not an array
+	 * @throws      \InvalidArgumentException       if the value contains objects, which are not an instance of UploadFile
+	 */
+	public function value($value) {
+		if (!is_array($value)) {
+			throw new \InvalidArgumentException('$value must be an array.');
+		}
+		
+		foreach ($value as $file) {
+			if (!($file instanceof UploadFile)) {
+				throw new \InvalidArgumentException('All given files must be an instance of '. UploadFile::class .'.');
+			}
+		}
+		
+		$this->registerField();
+		
+		UploadHandler::getInstance()->registerFilesByField($this->getId(), $value);
+	}
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function hasSaveValue() {
+		return false;
+	}
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function cleanup() {
+		$this->unregisterField();
+		
+		return $this;
+	}
+	
+	/**
+	 * @inheritDoc
+	 * 
+	 * @throws      \RuntimeException       if the field has already been initalisated
+	 */
+	public function maximum($maximum = null) {
+		if ($this->isRegistered()) {
+			throw new \LogicException('The upload field has already been registered. Therefore no modifications are allowed.');
+		}
+		
+		return $this->traitMaximum($maximum);
+	}
+	
+	/**
+	 * Sets the imageOnly flag for this field.
```

Comment:
Please improve the documentation of what the method does see documentation for `isImageOnly()`.